smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -105,11 +104,50 @@ private OffsetStore getOffsetStore(TopicPartition topicPartition) {
      */
     @Override
     public void partitionsAssigned(Set<TopicPartition> partitions) {
-        offsetStores.clear();
+        stopFlushAndCheckHealthTimer();
+
+        // Remove all handled partitions that are not in the given list of partitions
+        for (TopicPartition partition : new HashSet<>(offsetStores.keySet())) {
+            if (!partitions.contains(partition)) {
+                offsetStores.remove(partition);
+            }
+        }
+
+        // We cannot commit the removed offsets as we are not assigned to this partition anymore.
+
+        if (!offsetStores.isEmpty()) {
+            startFlushAndCheckHealthTimer();
+        }
+    }
 
+    /**
+     * Revoked partitions.
+     * This method is called from a Vert.x event loop (the one used by the Kafka client)
+     *
+     * @param partitions the partitions that we will no longer receive
+     */
+    @Override
+    public void partitionsRevoked(Set<TopicPartition> partitions) {
         stopFlushAndCheckHealthTimer();
 
-        if (!partitions.isEmpty()) {
+        // Remove all handled partitions that are not in the given list of partitions
+        Map<TopicPartition, OffsetAndMetadata> toCommit = new HashMap<>();
+        for (TopicPartition partition : new HashSet<>(offsetStores.keySet())) {
+            if (!partitions.contains(partition)) {
+                OffsetStore store = offsetStores.remove(partition);
+                long largestOffset = store.clearLesserSequentiallyProcessedOffsetsAndReturnLargestOffset();
+                if (largestOffset > -1) {
+                    toCommit.put(partition, new OffsetAndMetadata(largestOffset + 1L, null));
+                }
+            }
+        }
+
+        if (!toCommit.isEmpty()) {
+            // Commit the offsets
+            consumer.getDelegate().commit(toCommit);
+        }
```

Comment:
Concerning the comment left on the partitionsAssigned method. After committing we should remove partitions from the store.

Reply:
They have been already removed in `OffsetStore store = offsetStores.remove(partition);`
The commit happens after the removal.

Reply:
I see. nice!